### PR TITLE
Inline procedure framework

### PR DESCRIPTION
**Pull request by bobbybee**
Project example is 45115874
This implements a native substr block with a failsafe for Scratch / non-nativized Phosphorus, but it runs much faster with the flag set.

If this is accepted, I can do more primitives for operations like bitwise for example.

### CODE REVIEW

**Comment by bobbybee** [comment]
Solves #209

**Comment by bobbybee** [comment]
Any updates?

**Comment by nathan** [comment]
Haven't had time to review, sorry. I'll get to it.

File: index.html [hunk 1]
```diff
@@ -78,10 +78,10 @@ <h1 class=title>phosphorus</h1>
...
 </section>
 <script src=fonts.js></script>
-<script src=//cdnjs.cloudflare.com/ajax/libs/jszip/2.4.0/jszip.js></script>
-<script src=//canvg.googlecode.com/svn/trunk/rgbcolor.js></script>
-<script src=//canvg.googlecode.com/svn/trunk/StackBlur.js></script>
-<script src=//canvg.googlecode.com/svn/trunk/canvg.js></script>
+<script src=http://cdnjs.cloudflare.com/ajax/libs/jszip/2.4.0/jszip.js></script>
+<script src=http://canvg.googlecode.com/svn/trunk/rgbcolor.js></script>
+<script src=http://canvg.googlecode.com/svn/trunk/StackBlur.js></script>
+<script src=http://canvg.googlecode.com/svn/trunk/canvg.js></script>
```
Comment (tjvr) [inline index.html]: Why is this in here? :-)

**Comment by tjvr** [comment]
Is `native phosphorous inline %s` the best convention to use?

As I said in #209, I was imagining having a standard implementation for each custom block. They might be named eg. `%n << %n`, and then we'd recognise the scripts inside them as exactly-matching our implementation. What's your reasoning? :-)

File: index.html (hunk 1 above)
Comment (bobbybee) [inline index.html]: ...because it was the only way I managed to get phosphorus to work on my system?

Comment (tjvr) [inline index.html]: Should be omitted from this PR, regardless! :)

Comment (bobbybee) [inline index.html]: Yeah. Was a long time ago. Will just implement this properly in arsenic (it's my poisonous version of phopshorus haha)

**Pull request closed without merging**